### PR TITLE
Fixies profiling + conflict run pod sandbox

### PR DESCRIPTION
**Pull request by runcom**
@mrunalp @sameo PTAL

### CODE REVIEW

**Comment by mrunalp** [comment]
I think we should use a separate flag to enable profiling.

**Comment by runcom** [comment]
> I think we should use a separate flag to enable profiling.

suggestion? `--enable-profiler`? I still believe profiler must be enabled only in debug mode though

**Comment by mrunalp** [comment]
--profile

File: server/sandbox_run.go [hunk 1]
```diff
@@ -109,6 +109,9 @@ func (s *Server) RunPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 				return nil, err
 			}
 			dupID := matches[1]
+			if _, err := s.StopPodSandbox(ctx, &pb.StopPodSandboxRequest{PodSandboxId: dupID}); err != nil {
+				return nil, err
+			}
```
Comment (sameo) [inline server/sandbox_run.go]: Nice catch.

**Comment by runcom** [comment]
> I agree with @mrunalp here, most of the time you want to minimize your profiling overhead and coupling it with --debug could add a lot of noise. --profile sounds good to me.

mmm I'm not sure, is anyone enabling the profiler w/o being in debug? Profiler doesn't clutter the debug output at all, I just want to be safe and enable it just in debug.

**Comment by sameo** [comment]
@runcom it's debug that could potentially clutters the profiling results, making your profiling results less relevant.

**Comment by runcom** [comment]
> @runcom it's debug that could potentially clutters the profiling results, making your profiling results less relevant.

gotcha, was missing that (not sure how overhead debugs add but I trust you all :smile: )

**Comment by runcom** [comment]
Now under `--profile` PTAL guys

**Comment by sameo** [comment]
LGTM

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**